### DISCUSSION TITLE
Caddy proxy to access Home Assistant via Tailscale

### DISCUSSION THREAD

**Discussion by elcajon** [hassio-addons/app-tailscale#123] [Ideas]
I forked this repo a while ago to add some options like SSH and different subnets. Long story short, recently I came across [this announcement](https://tailscale.com/blog/caddy/) from Tailscale. 

_With the [beta release of Caddy 2.5](https://github.com/caddyserver/caddy/releases/tag/v2.5.0-beta.1), Caddy automatically recognizes and uses certificates for your Tailscale network (*.ts.net), and can use Tailscale’s HTTPS certificate provisioning when spinning up a new service._

This allows [Caddy servers](https://caddyserver.com/docs/quick-starts/reverse-proxy) proxy functionality to manage Tailscale certificates without any effort, making it possible to access Home Assistant securely (I know that the connection is secure even without HTTPS, as it is routed through Tailscale Wireguard VPN, but browsers aren't aware) via `https`.

The Caddy service coding itself is a no brainer:
https://github.com/elcajon-tech/addon-tailscale/blob/8541d83baf2193c2d5ed15ffe9ba2055822d669b/tailscale/rootfs/etc/services.d/caddy/run

Is there any interest in including the function in this add-on? I guess it will also solve the main use case described in #62 and #89.

**Comment by Mincka** [comment]
I guess people using https://github.com/tsujamin/hass-addons/tree/main/tailscale would be happy to have this.
Indeed, it's easier to use and to keep things running. It would help people having issues with manual certificate renewal.
I don't think it will be ever considered by the maintainer of this repo. Even if optional at runtime, it adds a dependency to a third-party in the image. It will increase the risk of bugs and make the support even more complicated...

I was wondering if the Caddy part could be independent and took a look at your code. I did not understand how you can enable or not the proxy mode. The option `enable_proxy` is in the translation file but not elsewhere.

**Reply by elcajon** [reply]
> I was wondering if the Caddy part could be independent and took a look at your code. I did not understand how you can enable or not the proxy mode. The option `enable_proxy` is in the translation file but not elsewhere.

I had used a configuration option for this in an earlier version, but have adjusted the logic to always enable Caddy proxy when HTTPS is active for the used Tailnet. I will remove the option from the translation in the next release - I simply forgot.



> I don't think it will be ever considered by the maintainer of this repo. Even if optional at runtime, it adds a dependency to a third-party in the image. It will increase the risk of bugs and make the support even more complicated...

That's why I started the discussion. If there is an interest I would open a PR, if not then not. :)

**Comment by frenck** [comment]
If you need/want a proxy, please use a proxy addon. It should not belong inside this addon. Indeed I probably won't add it in here.
